File: src/main/java/uk/co/alt236/tabclirun/ui/views/FontProvider.java
package uk.co.alt236.tabclirun.ui.views;

import java.awt.*;

class FontProvider {

    public String getMonospaceFontName() {
        return getFont("Book", Font.MONOSPACED);
    }

    private String getFont(final String fontName,
                           final String fallback) {
        final GraphicsEnvironment g = GraphicsEnvironment.getLocalGraphicsEnvironment();

        for (String font : g.getAvailableFontFamilyNames()) {
            //System.out.println(font);
            if (font.equals(fontName)) {
                return fontName;
            }
        }

        return fallback;
    }
}

File: app/src/main/java/uk/co/alt236/tabclirun/ui/views/common/FontProvider.kt
package uk.co.alt236.tabclirun.ui.views.common

import java.awt.Font
import java.awt.GraphicsEnvironment

internal class FontProvider {
    val monospaceFontName: String
        get() = getFont("Book", Font.MONOSPACED)

    private fun getFont(fontName: String, fallback: String): String {
        val g = GraphicsEnvironment.getLocalGraphicsEnvironment()

        for (font in g.availableFontFamilyNames) {
            //System.out.println(font);
            if (font == fontName) {
                return fontName
            }
        }

        return fallback
    }
}
